hooks/pre/30-upgrade_postgresql.rb

```diff
@@ -0,0 +1,18 @@
+def postgresql_10_upgrade
+  execute('foreman-maintain service start --only=postgresql')
+  (name, owner, enconding, collate, ctype, privileges) = %x{runuser postgres -c 'psql -lt | grep -E "^\s+postgres"'}.chomp.split('|').map(&:strip)
+  execute('foreman-maintain service stop')
+  execute('yum -y install rh-postgresql10-postgresql-server')
+  execute(%Q{scl enable rh-postgresql10 "PGSETUP_INITDB_OPTIONS='--lc-collate=#{collate} --lc-ctype=#{ctype} --locale=#{collate}' postgresql-setup --upgrade"})
+  execute('yum -y remove postgresql postgresql-server')
+  execute('rm -f /etc/systemd/system/postgresql.service')
+  execute('yum -y install rh-postgresql10-syspaths')
+end
+
+if local_postgresql? && rhel7? && module_enabled?('foreman')
+  `rpm -q rh-postgresql10-syspaths`
```

Comment:
Wouldn't be checking `/var/opt/rh/rh-postgresql10/lib/pgsql/data` exists be "cheaper" (no shell out to `rpm`)?

Reply:
Also, given this code exists in two places, should it be a helper function like `needs_postgresql_scl_upgrade?`